app/controllers/trips_controller.rb

```diff
@@ -0,0 +1,61 @@
+class TripsController < ApplicationController
+  def index
+    if params[:contoller] == "passengers"
+      @trips = Passenger.find_by(id: params[:id]).trips
+    elsif params[:controller] == "drivers"
+      @trips = Driver.find_by(id: params[:id]).trips
+    else
+      @trips = Trip.all
+    end
+  end
+
+  def show
+    @trip = Trip.find_by(id: params[:id])
+  end
+
+  def new
+    @trip = Trip.new
+    if params[:passenger_id]
+      @trip.passenger_id = params[:passenger_id]
+    end
+    @trip.driver_id = Driver.where(available: true).sample.id
+    @trip.date = Date.today
+    @trip.rating = nil
+  end
+
+  def create
+    @trip = Trip.new(trip_params)
+
+    if @trip.save
+      redirect_to passenger_path(@passenger)
```

Comment:
How would you have an `@passenger` attribute?  You need to do something like 

`redirect_to passenger_path(@trip.passenger.id)`